src/fs.cpp

```diff
@@ -148,7 +148,10 @@ std::string get_filesystem_error_message(const fs::filesystem_error& e)
 #ifdef __GLIBCXX__
 
 // reference: https://github.com/gcc-mirror/gcc/blob/gcc-7_3_0-release/libstdc%2B%2B-v3/include/std/fstream#L270
-
+#if defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wswitch"
+#endif
 static std::string openmodeToStr(std::ios_base::openmode mode)
```

Comment:
This will disable `-Wswitch` for any GCC compiler (even latest on Linux). Shouldn't we restrict that somehow? Maybe check if `WIN32` is defined or for a specific GCC version?

Reply:
https://github.com/bitcoin/bitcoin/blob/06ff47560ecb1d6a80e9fbd2f169f6ed8f31c464/src/fs.cpp#L147-L148

Reply:
Oh, it is already inside `#ifdef WIN32`, sorry for the noise.